stan/math/prim/fun/unit_vector_constrain.hpp

```diff
@@ -24,13 +24,16 @@ namespace math {
  */
 template <typename T, require_eigen_col_vector_t<T>* = nullptr,
           require_not_vt_autodiff<T>* = nullptr>
-inline plain_type_t<T> unit_vector_constrain(const T& y) {
+inline auto unit_vector_constrain(T&& y) {
   using std::sqrt;
   check_nonzero_size("unit_vector_constrain", "y", y);
-  const auto& y_ref = to_ref(y);
-  value_type_t<T> SN = dot_self(y_ref);
-  check_positive_finite("unit_vector_constrain", "norm", SN);
-  return (y_ref / sqrt(SN)).eval();
+  return make_holder(
+      [](const auto& y_ref) {
+        value_type_t<T> SN = dot_self(y_ref);
+        check_positive_finite("unit_vector_constrain", "norm", SN);
+        return y_ref / sqrt(SN);
+      },
+      to_forwarding_ref(std::forward<T>(y)));
```

Comment:
Why forward into the forwarding ref?

Reply:
If y is a rvalue reference to plain type, `to_forwarding_ref` returns a rvalue reference to y. To do that it needs to know whether y is a rvalue or lvalue. So we need `forward`.

Reply:
Actually ... no, given that we decided input arguments do not need holders that is not needed.